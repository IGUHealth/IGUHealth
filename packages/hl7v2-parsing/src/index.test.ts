import { test, expect } from "@jest/globals";
import parseHl7V2Message from "./index";
import fs from "node:fs";

test("ADT Message Parsing", () => {
  const a011 = fs.readFileSync("./test-data/adt-a01-1.txt", "utf8");
  const a012 = fs.readFileSync("./test-data/adt-a01-2.txt", "utf8");
  expect(parseHl7V2Message(a011)).toMatchSnapshot();
  expect(parseHl7V2Message(a012)).toMatchSnapshot();
});
