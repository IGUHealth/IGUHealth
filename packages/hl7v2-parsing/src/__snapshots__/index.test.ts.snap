// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADT Message Parsing 1`] = `
{
  "EVN": [
    "A01",
    "20200508130643",
    "",
    "",
    [
      "C006",
      "Wolf",
      "Kathy",
      "",
      "",
      "Dr",
      "",
      "",
      "DRNBR",
      "PRSNL",
      "",
      "",
      "ORGDR",
    ],
    "",
  ],
  "MSH": [
    "|",
    "^~\\&",
    "SIMHOSP",
    "SFAC",
    "RAPP",
    "RFAC",
    "20200508130643",
    "",
    [
      "ADT",
      "A01",
    ],
    "5",
    "T",
    "2.3",
    "",
    "",
    "AL",
    "",
    "44",
    "ASCII",
  ],
  "PD1": [
    "",
    "",
    [
      "FAMILY PRACTICE",
      "",
      "12345",
    ],
    "",
  ],
  "PID": [
    "1",
    [
      "2590157853",
      "",
      "",
      "SIMULATOR MRN",
      "MRN",
    ],
    [
      [
        "2590157853",
        "",
        "",
        "SIMULATOR MRN",
        "MRN",
      ],
      [
        "2478684691",
        "",
        "",
        "NHSNBR",
        "NHSNMBR",
      ],
    ],
    "",
    [
      "Esterkin",
      "AKI Scenario 6",
      "",
      "",
      "Miss",
      "",
      "CURRENT",
    ],
    "",
    "19890118000000",
    "F",
    "",
    "",
    [
      "170 Juice Place",
      "",
      "London",
      "",
      "RW21 6KC",
      "GBR",
      "HOME",
    ],
    "",
    [
      "020 5368 1665",
      "HOME",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    [
      "R",
      "Other - Chinese",
      "",
      "",
      "",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  "PV1": [
    "1",
    "I",
    [
      "RenalWard",
      "MainRoom",
      "Bed 1",
      "Simulated Hospital",
      "",
      "BED",
      "Main Building",
      "5",
    ],
    "28b",
    "",
    "",
    [
      "C006",
      "Wolf",
      "Kathy",
      "",
      "",
      "Dr",
      "",
      "",
      "DRNBR",
      "PRSNL",
      "",
      "",
      "ORGDR",
    ],
    "",
    "",
    "MED",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    [
      "6145914547062969032",
      "",
      "",
      "",
      "visitid",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "ARRIVED",
    "",
    "",
    "20200508130643",
    "",
    "",
  ],
}
`;

exports[`ADT Message Parsing 2`] = `
{
  "EVN": [
    "A01",
    "20200508130643",
    "",
    "",
    [
      "C006",
      "Wolf",
      "Kathy",
      "",
      "",
      "Dr",
      "",
      "",
      "DRNBR",
      "PRSNL",
      "",
      "",
      "ORGDR",
    ],
    "",
  ],
  "MSH": [
    "|",
    "^~\\&",
    "SIMHOSP",
    "SFAC",
    "RAPP",
    "RFAC",
    "20200508130643",
    "",
    [
      "ADT",
      "A01",
    ],
    "5",
    "T",
    "2.3",
    "",
    "",
    "AL",
    "",
    "44",
    "ASCII",
  ],
  "PD1": [
    "",
    "",
    [
      "FAMILY PRACTICE",
      "",
      "12345",
    ],
    "",
  ],
  "PID": [
    "1",
    [
      "2590157853",
      "",
      "",
      "SIMULATOR MRN",
      "MRN",
    ],
    [
      [
        "2590157853",
        "",
        "",
        "SIMULATOR MRN",
        "MRN",
      ],
      [
        "2478684691",
        "",
        "",
        "NHSNBR",
        "NHSNMBR",
      ],
    ],
    "",
    [
      "Esterkin",
      "AKI Scenario 6",
      "",
      "",
      "Miss",
      "",
      "CURRENT",
    ],
    "",
    "19890118000000",
    "F",
    "",
    "",
    [
      "170 Juice Place",
      "",
      "London",
      "",
      "RW21 6KC",
      "GBR",
      "HOME",
    ],
    "",
    [
      "020 5368 1665",
      "HOME",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    [
      "R",
      "Other - Chinese",
      "",
      "",
      "",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
  "PV1": [
    "1",
    "I",
    [
      "RenalWard",
      "MainRoom",
      "Bed 1",
      "Simulated Hospital",
      "",
      "BED",
      "Main Building",
      "5",
    ],
    "28b",
    "",
    "",
    [
      "C006",
      "Wolf",
      "Kathy",
      "",
      "",
      "Dr",
      "",
      "",
      "DRNBR",
      "PRSNL",
      "",
      "",
      "ORGDR",
    ],
    "",
    "",
    "MED",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    [
      "6145914547062969032",
      "",
      "",
      "",
      "visitid",
    ],
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "ARRIVED",
    "",
    "",
    "20200508130643",
    "",
    "",
  ],
}
`;
